examples/06-kitchen-sink/src/block/block.js

```diff
@@ -0,0 +1,482 @@
+		type: "string",
+		source: "html",
+		selector: "p"
+	},
+	exampleRichList: {
+		type: "string",
+		selector: "ul"
+	},
+	examplePlainText: {
+		type: "string"
+	},
+	exampleLink: {
+		type: "string"
+	},
+	exampleImageAlt: {
+		type: "string"
+	},
+	exampleImageURL: {
+		type: "string",
+		source: "attribute",
+		attribute: "src",
+		selector: "img"
+	},
+	exampleWidth: {
+		type: "string",
+		default: ""
+	}
+};
+
+/**
+ * Register: aa Gutenberg Block.
+ *
+ * Registers a new block provided a unique name and an object defining its
+ * behavior. Once registered, the block is made editor as an option to any
+ * editor interface where blocks are implemented.
+ *
+ * @link https://wordpress.org/gutenberg/handbook/block-api/
+ * @param  {string}   name     Block name.
+ * @param  {Object}   settings Block settings.
+ * @return {?WPBlock}          The block, if it has been successfully
+ *                             registered; otherwise `undefined`.
+ */
+registerBlockType("cgb/block-kitchen-sink", {
+	// Block name. Block names must be string that contains a namespace prefix. Example: my-plugin/my-custom-block.
+	title: __("kitchen-sink - CGB Block"), // Block title.
+	icon: "shield", // Block icon from Dashicons → https://developer.wordpress.org/resource/dashicons/.
+	category: "create-guten-block-kitchen", // Custom Block category: Look in src/init.php on Line 67. You can also use default groups E.g. common, formatting, layout widgets, embed.
+	keywords: [
+		__("kitchen-sink — CGB Block"),
+		__("CGB Example"),
+		__("create-guten-block")
+	],
+	supports: {
+		// Adds block supports: https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#supports-optional
+		align: true,
+		alignWide: true,
+		anchor: false,
+		customClassName: true,
+		className: true,
+		html: true,
+		inserter: true,
+		multiple: true,
+		reusable: true
+	},
+	attributes: kitchenAttributes, // Use attributes from line 42
+
+	/**
+	 * The edit function describes the structure of your block in the context of the editor.
+	 * This represents what the editor will render when the block is used.
+	 *
+	 * The "edit" property must be a valid function.
+	 *
+	 * @link https://wordpress.org/gutenberg/handbook/block-api/block-edit-save/
+	 */
+	edit: function(props) {
+		// Setup the attributes for the editor
+		const {
+			attributes: {
+				exampleToggle,
+				exampleRange,
+				exampleSelect,
+				exampleText,
+				exampleTextarea,
+				exampleColor,
+				exampleAlignment,
+				exampleRichH2,
+				exampleRichParagraph,
+				exampleRichList,
+				examplePlainText,
+				exampleLink,
+				exampleImage,
+				exampleImageURL,
+				exampleImageAlt
+			},
+			attributes,
+			isSelected,
+			editable,
+			className,
+			setAttributes
+		} = props;
+		// example options for SelectControl
+		const exampleOptions = [
+			{ value: "option_1", label: __("Option 1") },
+			{ value: "option_2", label: __("Option 2") }
+		];
+
+		return [
```

Comment:
Explain in details how the JSX going to be for the block. The real HTML and the Wrappers. Without this explanation example is not going to be helpful.